AzureAttestationExtension/agent.py

```diff
@@ -0,0 +1,708 @@
+#!/usr/bin/env python
+#
+# AzureAttestationLinuxAgent Extension
+#
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import os.path
+import signal
+import pwd
+import grp
+import re
+import sys
+import traceback
+import time
+import platform
+import subprocess
+import json
+import base64
+import inspect
+import shutil
+import crypt
+import xml.dom.minidom
+import re
+from distutils.version import LooseVersion
+
+from threading import Thread
+
+try:
+    from Utils.WAAgentUtil import waagent
+    import Utils.HandlerUtil as HUtil    
+    
+except Exception as e:
+    # These utils have checks around the use of them; this is not an exit case
+    print('Importing utils failed with error: {0}'.format(e))
+
+# Global Variables
+InitialRetrySleepSeconds = 30
+
+# https://github.com/Azure/azure-marketplace/wiki/Extension-Build-Notes-Best-Practices#error-codes-and-messages-output-to-stderr
+# Well known error codes for extnesion handlers. For azure attestation unsupported OS is important as we do not want to get flagged
+# if customer tries to run it on one of the unsupported distros.
+MissingorInvalidParameterErrorCode = 53
+UnsupportedOperatingSystem = 51
+
+# Configuration
+HUtilObject = None
+SettingsSequenceNumber = None
+HandlerEnvironment = None
+SettingsDict = None
+
+# Default configuration settings
+MaaEndpoint = "https://microsoft.attestation.net/",
+AscReportingEndpoint = "https://management.azure.com/",
+UseCustomeToken = "false"        
+    
+# Change permission of log path - if we fail, that is not an exit case
+# like any other extension attestation extension is started as a root so we should be good here.
+try:
+    ext_log_path = '/var/log/azure/'
+    if os.path.exists(ext_log_path):
+        os.chmod(ext_log_path, 700)
+except:
+    pass
+
+def main():
+    """
+    Main method
+    Parse out operation from argument, invoke the operation, and finish.
+    """
+    init_waagent_logger()
+    waagent_log_info('Azure Attestation Agent for Linux started to handle.')
+
+    # Determine the operation being executed
+    operation = None
+    try:
+        option = sys.argv[1]
+        if re.match('^([-/]*)(disable)', option):
+            operation = 'Disable'
+        elif re.match('^([-/]*)(uninstall)', option):
+            operation = 'Uninstall'
+        elif re.match('^([-/]*)(install)', option):
+            operation = 'Install'
```

Comment:
Why can't we use Argparse for parsing args here? They are more robust and more elegant.

Reply:
good suggestion. Bit newbie with python

Reply:
I have referred other python extension code and they are also written in the same way so this approach seems pretty much standard so I would suggest using it as it is.